src/nuxt-plugin.ts

```diff
@@ -21,18 +20,23 @@ const PlausiblePlugin: Plugin = (context, inject) => {
     ...context.$config?.plausible
   } as PlausibleModuleOptions
 
-  if (options.domain !== null) {
-    const plausible = Plausible(options)
-
-    if (options.enableAutoPageviews === true) {
-      plausible.enableAutoPageviews()
-    }
-    if (options.enableAutoOutboundTracking === true) {
-      plausible.enableAutoOutboundTracking()
+  if (optionsDomain.length) {
+    options = {
+      ...options,
+      domain: optionsDomain
     }
```

Comment:
I suggest to set the domain directly. This way, `options` can remain a constant.
```suggestion
    options.domain = optionsDomain
```

Reply:
While I'd like that too, the module options type decorate that property as read-only:

https://github.com/plausible/plausible-tracker/blob/master/src/lib/tracker.ts#L20

So in that case we'd have to create a separate type to cast to the plausible one which is arguably more complicated than it's worth in my opinion. Thoughts? 🙂